model_hub/examples/huggingface/question-answering/qa_beam_search_trial.py

```diff
@@ -0,0 +1,226 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import functools
+import logging
+from typing import Dict, Union
+
+import attrdict
+import data_beam_search
+import datasets
+import torch
+import transformers
+
+import determined.pytorch as det_torch
+import model_hub.huggingface as hf
+import model_hub.utils as utils
+
+
+class QABeamSearchTrial(hf.BaseTransformerTrial):
+    def __init__(self, context: det_torch.PyTorchTrialContext) -> None:
+        self.logger = logging.getLogger(__name__)
+        self.hparams = attrdict.AttrDict(context.get_hparams())
+        self.data_config = attrdict.AttrDict(context.get_data_config())
+        self.context = context
+
+        # Check to make sure the dataset is configured correctly.
+        if self.data_config.dataset_name is not None:
+            dataset_name = self.data_config.dataset_name
+            if dataset_name == "squad":
+                assert (
+                    not self.data_config.version_2_with_negative
+                ), "version_2_with_negative should be false for squad"
+            elif dataset_name == "squad_v2":
+                assert (
+                    self.data_config.version_2_with_negative
+                ), "version_2_with_negative should be true for squad_v2"
+
+        self.data_processors = data_beam_search
+
+        # Get the datasets: you can either provide your own CSV or JSON training and evaluation
+        # files (see below) or just provide the name of one of the public datasets available on the
+        # hub at https://huggingface.co/datasets/ (the dataset will be downloaded automatically
+        # from the datasets Hub).
+
+        # For CSV/JSON files, this script will use the column called 'text' or the first column if
+        # no column called 'text' is found. You can easily tweak this behavior (see below).
+
+        # See more about loading any type of standard or custom dataset (from files, python dict,
+        # pandas DataFrame, etc) at
+        # https://huggingface.co/docs/datasets/loading_datasets.html.
+        self.raw_datasets = hf.default_load_dataset(self.data_config)
+        self.column_names = self.raw_datasets["train"].column_names
+
+        # For beam search, we need to use a different model from the default model returned by
+        # AutoModelForQuestionAnswering.  We will use a custom init in this case that is a slight
+        # modification of the BaseTransformerTrial init method.
+        self.exp_config = attrdict.AttrDict(context.get_experiment_config())
+
+        # Check to make sure all expected hyperparameters are set.
+        self.check_hparams()
```

Comment:
Why do we need this exactly? Don't really see it used anywhere else.

Reply:
this is called in `BaseTransformerTrial` to make sure all required hyperparameters are set.  the same checks apply for this trial.